packages/amplify_api/ios/Classes/FlutterGraphQLApi.swift

```diff
@@ -131,4 +131,74 @@ class FlutterGraphQLApi {
+                                        print("Subscription established: \(id)")
+                                        flutterResult(nil)
+                                    }
+                                case .data(let result):
+                                    switch result {
+                                    case .success(let data):
+                                        print("GraphQL subscription event received: \(data)")
+                                        let payload: [String: Any] = [
+                                            "data": data,
+                                            "errors": []
+                                        ]
+                                        graphQLSubscriptionsStreamHandler.sendEvent(payload: payload, id: id, type: GraphQLSubscriptionEventTypes.DATA)
+                                    case .failure(let error):
+                                        //TODO: Handle the different error responses
+                                        print("Got failed result with \(error.errorDescription)")
+                                    }
+                                }
+                            }) { result in
+                                switch result {
+                                case .success:
+                                    if(!id.isEmpty){
+                                        OperationsManager.removeOperation(cancelToken: id)
+                                    }
+                                    print("Subscription has been closed successfully")
+                                    graphQLSubscriptionsStreamHandler.sendEvent(payload: nil, id: id, type: GraphQLSubscriptionEventTypes.DONE)
+                                case .failure(let apiError):
+                                    print("Subscription has terminated with \(apiError)")
+                                    if(!id.isEmpty){
+                                        OperationsManager.removeOperation(cancelToken: id)
+                                    }
+                                    if(established) {
+                                        let errorMap = FlutterApiError.createErrorMap(localizedError: apiError.errorDescription, recoverySuggestion: apiError.errorDescription)
+                                        let flutterError = FlutterError(
+                                            code: "AmplifyException",
+                                            message: FlutterApiErrorMessage.SUBSCRIBE_FAILED.rawValue,
+                                            details: errorMap)
+                                        graphQLSubscriptionsStreamHandler.sendError(flutterError: flutterError)
+                                    } else {
+                                        FlutterApiError.handleAPIError(flutterResult: flutterResult, error: apiError, msg: FlutterApiErrorMessage.ESTABLISH_SUBSCRIPTION_FAILED.rawValue)
```

Comment:
That will not happen because we are setting the `established` flag to `true` before calling `flutterResult` in the success case.
In the case of an exception, we call `flutterResult` only if `established` is still false, and the subscription connection did not establish itself. If the subscription is established, and exceptions are encountered after that, they are sent through the `graphQLSubscriptionsStreamHandler`